zulip_bots/zulip_bots/test_lib.py

```diff
@@ -70,29 +71,58 @@ def check_expected_responses(self, expectations, expected_method='send_reply',
                 message = {'content': m, 'type': "private", 'display_recipient': recipient,
                            'sender_email': email, 'sender_id': sender_id,
                            'sender_full_name': sender_full_name}
-                self.assert_bot_response(message=message, response=response,
-                                         expected_method=expected_method, state_handler=state_handler)
+                self.assert_bot_responses(message, (response, expected_method),
+                                          state_handler=state_handler)
             if type != "private":
                 message = {'content': m, 'type': "stream", 'display_recipient': recipient,
                            'subject': subject, 'sender_email': email, 'sender_id': sender_id,
                            'sender_full_name': sender_full_name}
-                self.assert_bot_response(message=message, response=response,
-                                         expected_method=expected_method, state_handler=state_handler)
+                self.assert_bot_responses(message, (response, expected_method),
+                                          state_handler=state_handler)
 
-    def call_request(self, message, expected_method, response, state_handler):
-        # type: (Dict[str, Any], str, Dict[str, Any], Optional[StateHandler]) -> None
+    def call_request(self, message, *responses, **kwargs):
+        # type: (Dict[str, Any], *Tuple[Dict[str, Any], str], **Any) -> None
+        state_handler = kwargs.pop('state_handler', None)
         if state_handler is None:
             state_handler = StateHandler()
-        # Send message to the concerned bot
-        self.message_handler.handle_message(message, self.MockClass(), state_handler)
 
-        # Check if the bot is sending a message via `send_message` function.
-        # Where response is a dictionary here.
+        # Mock BotHandler; get instance
+        bot_handler = self.MockClass()
         instance = self.MockClass.return_value
-        if expected_method == "send_message":
-            instance.send_message.assert_called_with(response)
-        else:
-            instance.send_reply.assert_called_with(message, response['content'])
+
+        # Send message to the bot
+        try:
+            self.message_handler.handle_message(message, bot_handler, state_handler)
+        except KeyError as key_err:
+            raise Exception("Message tested likely required key {}.".format(key_err))
```

Comment:
I added this test, since some of my bots (maybe pollbot, possibly others) depended on parameters of message, which were not supplied in the version provided in the test. So, the KeyError is a hint to a test-writer that an additional key should be added to the test message. I'm fully open to having better descriptive text here, but I believe that's the case it is attempting to catch.

Of course, KeyError could arise due to other reasons, so this could be at least 'perhaps' rather than 'likely'?